src/service.py

```diff
@@ -67,6 +72,14 @@ def snap_install(resource: str) -> SnapService:
         return SnapService(snap_client)
 
 
+def remove_upstream_snap() -> None:
+    """Remove the old snap from upstream to not override with the charmed-openstack-exporter."""
+    try:
+        snap.remove(["golang-openstack-exporter"])
+    except snap.SnapError as e:
+        logger.error("failed to remove snap: %s", str(e))
```

Comment:
I don't think we should ignore this error; if the old snap fails to remove, the charm should block.  It's not a good state to have both installed at once.